src/graph.cpp

```diff
@@ -216,33 +256,50 @@ std::vector<std::set<node_id_t>> Graph::boruvka_emulation() {
   retval.reserve(temp.size());
   for (const auto& it : temp) retval.push_back(it.second);
 
+  if (make_copy) {
+    if(copy_in_mem) {
+      // restore original supernodes and free memory
+      for (node_id_t i = 0; i < num_nodes; i++) {
+        if (supernodes[i] != nullptr) free(supernodes[i]);
+        supernodes[i] = copy_supernodes[i];
```

Comment:
why can't we just put `supernodes = copy_supernodes`?

Reply:
Because some of the copy_supernodes are nullptrs and because we want to loop through to free the memory anyway. This was super rushed though so I'm welcome to better ideas

Reply:
ah gotcha. i think it would be a better idea to follow the same pattern for in-memory as we do on disk. that is, copy the sketches we will modify to storage and restore those sketches rather than working on the copy and pointing all the supernodes to the new-but-also-not-new sketches.

Reply:
Took a stab at doing this in the second most recent commit